src/go/rpk/pkg/cli/cmd/redpanda/start.go

```diff
@@ -1066,6 +1076,33 @@ func getOrFindInstallDir(fs afero.Fs, installDir string) (string, error) {
 	return foundConfig, nil
 }
 
+// equalConfigs checks if two config objects are equal.
+// It takes two inputs:
+//   - initRaw: a byte slice representation of the initial config.
+//   - cfg: a pointer to the current config.
+//
+// The function returns a boolean indicating if the two config objects are equal.
+func equalConfigs(initRaw []byte, cfg *config.Config) bool {
```

Comment:
Should we move this to within the `Config.Write` method, i.e. always avoid rewriting if the results are identical?

Reply:
Done, I created a new config.Config method and use it before Write.